modulus/utils/neighbor_list.py

```diff
@@ -0,0 +1,177 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple, Union
+
+import torch
+import warp as wp
+
+
+@wp.kernel
+def radius_search_count(
+    hashgrid: wp.uint64,
+    points: wp.array(dtype=wp.vec3),
+    queries: wp.array(dtype=wp.vec3),
+    result_count: wp.array(dtype=wp.int32),
+    radius: wp.float32,
+):
+    """
+    Warp kernel for counting the number of points within a specified radius
+    for each query point, using a hash grid for spatial queries.
+
+    Args:
+        hashgrid: An array representing the hash grid.
+        points: An array of points in space.
+        queries: An array of query points.
+        result_count: An array to store the count of neighboring points within the radius for each query point.
+        radius: The search radius around each query point.
+    """
+    tid = wp.tid()
+
+    # create grid query around point
+    qp = queries[tid]
+    query = wp.hash_grid_query(hashgrid, qp, radius)
+    index = int(0)
```

Comment:
Casting is redundant, since `0` is already `int` type. Unless it's some `warp` quirk? Same in other places.

Reply:
I tested this and apparently `int` is necessary